netbout-web/src/main/java/com/netbout/rest/bout/TkAttach.java

```diff
@@ -137,12 +138,31 @@ public Response act(final Request req) throws IOException {
     }
 
     /**
-     * Extracts file name.
+     * Reads the filename either from the name field or from the uploaded file.
+     * @param form From
+     * @param file File
+     * @return String name
+     * @throws IOException If fails
+     */
+    private String filename(final Smart form, final Request file)
+        throws IOException {
+        String name = null;
```

Comment:
@krzyk Actually the name should be taken from the file if the name given by the name part is null or empty. Would this be better then the current solution?

```
final String name;
Iterable<Request> reqs = form.part("name");
if (reqs.hasNext() && !StringUtils.isEmpty(IOUtils.toString(.....))) {
    name = IOUtils.toString.....;
} else {
   name = this.nameFromFile....;
}
```